src/HttpClient/HttpClientFactory.php

```diff
@@ -111,7 +112,7 @@ public function create(Options $options): HttpAsyncClientInterface
      */
     private function resolveClient(Options $options)
     {
-        if (class_exists(SymfonyHttplugClient::class)) {
+        if (class_exists(SymfonyPsr18Client::class) || class_exists(SymfonyHttplugClient::class)) {
```

Comment:
While this might be more of a hypothetical concern if, for any reason, we do end up with the `SymfonyHttplugClient` or if we fall through and `php-http/discovery defaults to `SymfonyHttplugClient`, we run into the same issue we're trying to solve.

Again, our stance is that we can accept living with an abandoned package for a bit longer. Causing exceptions, even if they do not stem from the SDK, is not ok though.

Reply:
I can easily remove everything related to the Httplug client, the PSR-18 client is more than enough as explained above. Do you feel safer this way?

Reply:
Not for the `php-http/discovery` case. Let's keep it for the moment. Maybe we get some feedback in #1541

Reply:
>Not for the php-http/discovery case. 

We can technically replace that case with the PSR-18 discovery. The only thing that would change is that we have to require `psr/http-client-implementation` instead of `php-http/async-client-implementation`. I may think about an unlikely case where you will not be able to update quietly to the new version because you are missing installed a PSR-18 compatible client, but as far as I see every single supported HTTP client by the autodiscovery that would be resolved now is also PSR-18 compatible, so there should be nothing to do anything: if it worked before, it should work next.